### PR TITLE
refactor: Move `iroh-router` into `iroh-net` and rename `iroh-net` to `iroh`

### PR DESCRIPTION
**Pull request by matheus23**
## Description

This merges `iroh-net`, `iroh-router` and `iroh` into simply one crate: `iroh`.

## Breaking Changes

- `iroh-net` got renamed to `iroh`
- `iroh-router` moved into `iroh`
- `iroh-router::ProtocolHandler` is now `iroh::protocol::ProtocolHandler`
- `iroh-router::ProtocolMap` is now `iroh::protocol::ProtocolMap`
- `iroh-router::Router` is now `iroh::router::Router`
- `iroh-router::RouterBuilder` is now `iroh::router::RouterBuilder`

## Notes & open questions

Minor Q: Do the `iroh::protocol` and `iroh::router` modules make sense?

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

**Comment by matheus23** [comment]
I highly recommend viewing this PR via its two individual commits.

**Comment by matheus23** [comment]
FWIW: If we want to merge this for real, we should make sure that both these PRs are rebased on main. There's some changes to e.g. the README missing on this branch. It's likely they'd get lost if we're not careful about rebasing.

**Comment by dignifiedquire** [comment]
> Do the `iroh::protocol` and `iroh::router` modules make sense?

what about 

- `iroh::protocol::ProtocolHandler`
- `iroh::protocol::ProtocolMap`
- `iroh::protocol::Router`
- `iroh::protocol::RouterBuilder`

**Comment by matheus23** [comment]
Yeah I like that better

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ env:
   RUSTFLAGS: -Dwarnings
   RUSTDOCFLAGS: -Dwarnings
   SCCACHE_CACHE_SIZE: "50G"
-  CRATES_LIST: "iroh,iroh-node-util,iroh-metrics,iroh-net,iroh-net-bench,iroh-test,iroh-dns-server,iroh-router,netwatch,portmapper,iroh-relay,iroh-net-report"
+  CRATES_LIST: "iroh,iroh-node-util,iroh-metrics,,iroh-net-bench,iroh-test,iroh-dns-server,netwatch,portmapper,iroh-relay,iroh-net-report"
```
Comment (dignifiedquire) [inline .github/workflows/tests.yaml:26]: double `,,`

File: iroh/Cargo.toml [hunk 2]
```diff
@@ -3,11 +3,11 @@ name = "iroh"
 version = "0.28.1"
 edition = "2021"
 readme = "README.md"
-description = "A toolkit for building distributed applications"
+description = "networking support for iroh"
```
Comment (dignifiedquire) [inline iroh/Cargo.toml:6]: wrong description

File: iroh/README.md [hunk 3]
```diff
@@ -1,25 +1,68 @@
-# iroh
+# iroh-net
```
Comment (dignifiedquire) [inline iroh/README.md:1]: this readme needs work, but we can do that as a follow up

File: iroh/bench/src/bin/bulk.rs [hunk 4]
```diff
@@ -35,12 +35,12 @@ pub fn run_iroh(opt: Opt) -> Result<()> {
         // enable recording metrics
         iroh_metrics::core::Core::try_init(|reg, metrics| {
             use iroh_metrics::core::Metric;
-            metrics.insert(iroh_net::metrics::MagicsockMetrics::new(reg));
-            metrics.insert(iroh_net::metrics::NetReportMetrics::new(reg));
-            metrics.insert(iroh_net::metrics::PortmapMetrics::new(reg));
+            metrics.insert(::iroh::metrics::MagicsockMetrics::new(reg));
```
Comment (dignifiedquire) [inline iroh/bench/src/bin/bulk.rs:38]: why `::`?

File: iroh/src/lib.rs [hunk 5]
```diff
@@ -1,19 +1,260 @@
-//! Send data over the internet.
+//! Peer-to-peer QUIC connections.
 //!
+//! iroh-net is a library to establish direct connectivity between peers.  It exposes an
```
Comment (dignifiedquire) [inline iroh/src/lib.rs:3]: iroh 😅

**Comment by matheus23** [comment]
Github is dorky. The branch was never deleted.
This PR moved to https://github.com/n0-computer/iroh/pull/2973

**Pull request closed without merging**